### PR TITLE
Know your neighborhood

### PR DESCRIPTION
**Pull request by moringspeaker**
I wrote a python script named **_location.py_** in a new directory GIS. In this directory, I stored a copy of NYC_GIS data which is in form of geojson file. This script can get the nearest neighborhood the user belong  according their lattitude and longtitude coordinates by searching through the geojson file. But I didn't intergrete this script into our backend. I also wrote a relative JS script called **_location.js_** located in directory **stastic/js**, this script is introduced in our **base.html**. What this script does is to ask if we can use user's geographic information at the first time user open it. Once the user agree. this script will get user's geo-information and pass it to backend. But I haven't change our backend's view function (index_view) yet. I guess we can add this geo feature as a new feature to our model and then we can display it everytime user open our website.

### CODE REVIEW

File: node_modules/.package-lock.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (vchrombie) [inline node_modules/.package-lock.json:1]: This file is not really needed to be added. Please add the `node_modules` to the `.gitignore` file.

File: GIS/location.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import json
```
Comment (vchrombie) [inline GIS/location.py:1]: Can you please move these scripts to the `scipts/` folder to keep these at once place?

File: static/CACHE/css/base.2ff388b33c76.css [hunk 3]
```diff
@@ -0,0 +1,456 @@
+/* Global Variables */
```
Comment (vchrombie) [inline static/CACHE/css/base.2ff388b33c76.css:1]: We are not tracking any of these `CACHE` files. Please pull the latest changes from develop and delete this folder.

File: users/templates/users/index.html [hunk 4]
```diff
@@ -1,10 +1,8 @@
 {% extends "base.html" %} {% block content %}
-
```
Comment (vchrombie) [inline users/templates/users/index.html:2]: Maybe you can get rid of these changes.

File: users/templates/users/login.html [hunk 5]
```diff
@@ -1,5 +1,4 @@
 {% extends "base.html" %} {% block content %} {% load bootstrap_icons %}
-
```
Comment (vchrombie) [inline users/templates/users/login.html:2]: Same here

File: GIS/Neighborhood Names GIS.geojson [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (vchrombie) [inline GIS/Neighborhood Names GIS.geojson:1]: Can you prettify this JSON maybe?

**Review by vchrombie — CHANGES_REQUESTED** [review]
I have added some comments to improve the changes. Please let me know if you need help with anything.

File: GIS/Neighborhood Names GIS.geojson (hunk 6 above)
Comment (vchrombie) [reply GIS/Neighborhood Names GIS.geojson:1]: You can maybe move this file to `scripts/datasets` if that makes sense.

**Comment by vchrombie** [comment]
Also, the build failed because of the issues in black formatting. Please resolve them as well so that we can merge this PR.

File: GIS/location.py (hunk 2 above)
Comment (moringspeaker) [reply GIS/location.py:1]: Sure

**Comment by moringspeaker** [comment]
I also installed a package named geopy using pip install, I'm trying do add it in poetry

File: GIS/location.py (hunk 2 above)
Comment (moringspeaker) [reply GIS/location.py:1]: Where is this folder

File: users/templates/users/index.html (hunk 4 above)
Comment (moringspeaker) [reply users/templates/users/index.html:2]: I didn't find any blank line on my local IDE
image
That's weird

**Comment by vchrombie** [comment]
> I also installed a package named geopy using pip install, I'm trying do add it in poetry

The build didn't fail because of this I guess but it failed because of the formatting and linter issues.

File: GIS/Neighborhood Names GIS.geojson (hunk 6 above)
Comment (vchrombie) [reply GIS/Neighborhood Names GIS.geojson:1]: Get rid of the change means I said you need not remove the line. You can maybe revert the change or add back the new line and add the change.

File: users/templates/users/index.html (hunk 4 above)
Comment (vchrombie) [reply users/templates/users/index.html:2]: Get rid of the change means I said you need not remove the line. You can maybe revert the change or add back the new line and add the change.

**Comment by StephanieHou** [comment]
@moringspeaker Closing this PR since we have restructured. Please submit a new PR with updated code.

**Pull request closed without merging**